### PR TITLE
Improve request handling in bufferserver when a chain of handlers are used

### PR DESCRIPTION
**Pull request by PramodSSImmaneni**
During request processing, not processing remaining data in a client by discarding it when processing is being passed from one client to the next in the chain

### CODE REVIEW

File: bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java [hunk 1]
```diff
@@ -441,7 +435,9 @@ public int readSize()
            * unregister the unidentified client since its job is done!
            */
           unregistered(key);
-          ignore = true;
+          // Control is being transferred to next client in the chain so no more processing in this client after this
```
Comment (243826) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java]: Pramod, this is different use case. Ignore is still needed. Subscriber is supposed to be ignored for the rest of its life.

File: bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java [hunk 2]
```diff
@@ -374,21 +368,18 @@ public void onMessage(byte[] buffer, int offset, int size)
         client.transferBuffer(buffer, readOffset + size, len);
       }
 
-      ignore = true;
+      // Remaining data has been transferred to next client in the chain and is going to be processed there so we would
```
Comment (243826) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java]: Java coding convention for comments as well 😀

File: bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java (hunk 1 above)
Comment (PramodSSImmaneni) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java]: Looks like the unregistered client is being unregistered in subscribe case as well and not used again. Is ignore still needed for rest of life, possible discard itself doesnt do much as there is no other data to be read but wouldn't it be safer and work as well.

File: bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java (hunk 2 above)
Comment (PramodSSImmaneni) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java]: Updated

File: bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java (hunk 1 above)
Comment (243826) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java]: true. my bad.

**Pull request merged**